### PR TITLE
jfr2pprof: Add allocations and lock data

### PR DESCRIPTION
**Pull request by NeQuissimus**
This is basically a follow-up to #587 

The previous PR was limited to converting CPU measurements, this one is adding allocations and contended locks.
Running the converter will create a file for each type of measurement, so that JFR files containing mixed data are untangled.

/cc @federico-quintas

### CODE REVIEW

**Comment by NeQuissimus** [comment]
@apangin I think this is good to go now

File: src/converter/jfr2pprof.java [hunk 1]
```diff
@@ -56,8 +60,20 @@ public boolean equals(final Object other) {
...
+
     public static final byte[] METHOD_UNKNOWN = "[unknown]".getBytes();
 
+    public static final byte[] VALUE_TYPE_CPU = "cpu".getBytes(StandardCharsets.UTF_8);
+    public static final byte[] VALUE_TYPE_ALLOC = "allocations".getBytes(StandardCharsets.UTF_8);
+    public static final byte[] VALUE_TYPE_LOCK = "lock".getBytes(StandardCharsets.UTF_8);
+
+    public static final byte[] VALUE_UNIT_CPU = "nanoseconds".getBytes(StandardCharsets.UTF_8);
+    public static final byte[] VALUE_UNIT_ALLOC = "count".getBytes(StandardCharsets.UTF_8);
+    public static final byte[] VALUE_UNIT_LOCK = "nanoseconds".getBytes(StandardCharsets.UTF_8);
```
Comment (apangin) [inline src/converter/jfr2pprof.java:75]: How come the value is "count" for allocations, but "nanoseconds" for locks? I see the converter code handles both events exactly the same way.
Also, does it make sense to include "bytes" in the allocation profile?

File: src/converter/jfr2pprof.java [hunk 2]
```diff
@@ -56,8 +60,20 @@ public boolean equals(final Object other) {
...
 
+    public static final String TYPE_CPU = "cpu";
+    public static final String TYPE_ALLOC = "alloc";
+    public static final String TYPE_LOCK = "lock";
+
     public static final byte[] METHOD_UNKNOWN = "[unknown]".getBytes();
 
+    public static final byte[] VALUE_TYPE_CPU = "cpu".getBytes(StandardCharsets.UTF_8);
+    public static final byte[] VALUE_TYPE_ALLOC = "allocations".getBytes(StandardCharsets.UTF_8);
+    public static final byte[] VALUE_TYPE_LOCK = "lock".getBytes(StandardCharsets.UTF_8);
```
Comment (apangin) [inline src/converter/jfr2pprof.java:71]: These fields (as well as the following ones) are redundant: they are never reused in the code.

File: src/converter/jfr2pprof.java [hunk 3]
```diff
@@ -209,19 +244,49 @@ private byte[] getMethodName(final long methodId, final byte methodType) {
...
-             final FileOutputStream out = new FileOutputStream(dst)) {
-            new jfr2pprof(jfr).dump(out);
+        if (types.isEmpty()) {
+            types.add(TYPE_CPU);
+        }
+
+        for (final String type : types) {
+          File dst = new File(arguments.output + "." + type);
+
+          if (dst.isDirectory()) {
```
Comment (apangin) [inline src/converter/jfr2pprof.java:282]: This code was intended for a case when a user specifies a directory as an output argument. However, here `dst` has already been altered with the type suffix.

File: src/converter/jfr2pprof.java [hunk 4]
```diff
@@ -209,19 +244,49 @@ private byte[] getMethodName(final long methodId, final byte methodType) {
...
+              dst = new File(dst, new File(arguments.input).getName().replace(".jfr", ".pprof" + "." + type));
```
Comment (apangin) [inline src/converter/jfr2pprof.java:283]: I'd put `type` ahead of the file extension and only for the case when multiple types are given.

**Comment by apangin** [comment]
Hi, is this PR still relevant?

**Comment by NeQuissimus** [comment]
It's very relevant as the functionality otherwise does not exist. I have not had time to update this but will try to get to it.

**Comment by apangin** [comment]
OK, no worries. I will leave the PR open and review it when updated. Thank you.

**Comment by apangin** [comment]
Obsoleted by #905

**Pull request closed without merging**